### ISSUE TITLE
Expense - Undefined displayed in violation message instead of amount when tax enabled

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#41771]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.71.0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
recondition: user is a member of a collect policy with enabled taxes
1. Go to https://staging.new.expensify.com/
and log in as member
2. Navigate to the workspace chat and submit expense > $200
3. Submit the report
4. Log in as the admin
5. Navigate to the workspace chat
6. Open the submitted report and wait for violations to appear

## Expected Result:
The violation message includes exceeding amount

## Actual Result:
The violation message "Expense exceeds auto approval limit of undefined" is displayed

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/645607c2-070d-4cd2-991a-5d85dab98126

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
@kevinksullivan  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by kevinksullivan** [comment]
feels like collect polish to me, so moving it to that project.

**Comment by roitman-g** [comment]
Could anyone provide the steps for enabling the recondition - "user is a member of a collect policy with enabled taxes"?

**Comment by cead22** [comment]
You're right. I'll submit a back end fix to send `formattedLimit` in the data

**Comment by cead22** [comment]
Looking closer, this is a notice that is being shown under the receipt, because I think we're showing all notices in that place, but maybe what we need to do here right now is to not show that. What do you think @JmillsExpensify 

Also, I think the setup for this isn't about taxes, but about having auto approval enabled and an auto approval limit set

**Comment by cead22** [comment]
The RBR on the money request preview looks right in the video in the issue description, because it would have the receipt required violation.

I think what we need to do is update `src/components/ReportActionItem/MoneyRequestView.tsx`

```js
// current
const noticeTypeViolations = transactionViolations?.filter((violation) => violation.type === 'notice').map((v) => ViolationsUtils.getViolationTranslation(v, translate)) ?? [];

// new
const receiptNotices = ['receiptRequired', 'receiptNotSmartScanned', 'modifiedDate', 'cashExpenseWithNoReceipt', 'smartscanFailed'];
const noticeTypeViolations = transactionViolations?.filter((violation) => receiptNotices.includes(violation.name)).map((v) => ViolationsUtils.getViolationTranslation(v, translate)) ?? [];
```

That way we're "whitelisting" all the possible violations that can go under the receipt, and shouldn't get surprised by new things appearing in there

**Comment by cead22** [comment]
@gijoe0295 @fedirjh what do you think ^?

**Comment by cead22** [comment]
> * Why is `modifiedDate` included there?

Because this is the name for the violation "Date differs from scanned receipt", so it makes sense to me to display it under the receipt

> 2\. The second point in my proposal also has the similar idea where we can filter both the violation type and the field so the other notices would show under their corresponding fields. By that we can leverage the violation-field mapping here:

Yeah I think that's a better way to implement it, can you update your proposal to functionally do what I'm suggesting. 

> * Fix the BE

I submitted a PR for this

> 2\. I think there might be confusion here when we want to show the `notice` violation in both receipt audit and amount field. I want to confirm expectation on this but if so, we can:

We don't wanna show the same violation/notice in two separate fields, so I don't think we need a new param, and we only need to update the mapping so `'receiptRequired', 'receiptNotSmartScanned', 'modifiedDate', 'cashExpenseWithNoReceipt', 'smartscanFailed'` map to the receipt and not amount or date or any other field. 

I think that would work if we decide to start showing all the notices, what do you think about this @JmillsExpensify?

**Comment by JmillsExpensify** [comment]
> We don't wanna show the same violation/notice in two separate fields, so I don't think we need a new param, and we only need to update the mapping so 'receiptRequired', 'receiptNotSmartScanned', 'modifiedDate', 'cashExpenseWithNoReceipt', 'smartscanFailed' map to the receipt and not amount or date or any other field.

Hmm, not sure. So looking at the design doc, these are the additional notes that could show. 

![image](https://github.com/Expensify/App/assets/1127863/7b9eb649-3c81-4b66-aae2-09197bb1ca81)

The only other note that we're bringing to NewDot is the last one for 7-day hold. It's two PRs away from being implemented. Tax we are not bringing to NewDot. Invoice markup is not supported. Then the note about a broken bank connection can't be fixed in NewDot. Then auto-approval is used by a super small percentage of customers. 

So I think I land in an opposite spot. I don't think we should show any of these notes aside from the 7-day hold note which is currently being worked on.

**Comment by cead22** [comment]
> The second point in my proposal also has the similar idea where we can filter both the violation type and the field so the other notices would show under their corresponding fields

@gijoe0295 let's not do this for now then, and do what I proposed [here](https://github.com/Expensify/App/issues/41771#issuecomment-2121308580), and update the mapping so `'receiptRequired', 'receiptNotSmartScanned', 'modifiedDate', 'cashExpenseWithNoReceipt', 'smartscanFailed'` map to the receipt and not amount or date or any other field, for good measure.

If that sounds good, can you submit a proposal? Thanks

**Comment by JmillsExpensify** [comment]
Not sure if ignore means that it's not on the whitelist or not, though generally, I don't think we should show that note in NewDot.

**Comment by cead22** [comment]
Should we remove this line instead `{canUseViolations && <ViolationMessages violations={getViolationsForField('receipt')} />}`, and that way we don't need `violation.type === CONST.VIOLATION_TYPES.NOTICE &&`?

**Comment by cead22** [comment]
`modifiedDate` is a notice

**Comment by kevinksullivan** [comment]
Hey @fedirjh can you finish out the steps above?

**Comment by kevinksullivan** [comment]
Payments made, just waiting on @fedirjh for final steps and then we're all set

**Comment by kevinksullivan** [comment]
Moving to weekly since we're only held on steps here.
